farm-boosted-rewards-subscriber/src/claim_farm_boosted.rs

```diff
@@ -1,23 +1,82 @@
 multiversx_sc::imports!();
 multiversx_sc::derive_imports!();
 
-mod claim_farm_boosted_rewards_proxy {
-    multiversx_sc::imports!();
+use auto_farm::common::address_to_id_mapper::{AddressId, AddressToIdMapper};
 
-    #[multiversx_sc::proxy]
-    pub trait ClaimFarmBoostedRewardsProxy {
-        #[endpoint(claimBoostedRewards)]
-        fn claim_boosted_rewards(&self, user: ManagedAddress) -> EsdtTokenPayment<Self::Api>;
+#[multiversx_sc::module]
+pub trait ClaimFarmBoostedRewardsModule:
+    subscriber::common_storage::CommonStorageModule + energy_query::EnergyQueryModule
+{
+    #[only_owner]
+    #[endpoint(addFarm)]
+    fn add_farm(&self, farm_address: ManagedAddress) -> AddressId {
+        require!(
+            self.blockchain().is_smart_contract(&farm_address),
+            "Invalid farm address"
+        );
+
+        self.farm_id().insert_new(&farm_address)
     }
-}
 
-#[derive(ManagedVecItem, Clone)]
-pub struct AdditionalFarmData {
-    pub dummy_data: u8,
-}
+    #[only_owner]
+    #[endpoint(removeFarm)]
+    fn remove_farm(&self, farm_address: ManagedAddress) -> AddressId {
+        self.farm_id().remove_by_address(&farm_address)
+    }
+
+    #[endpoint(performClaimRewardsOperations)]
+    fn perform_claim_rewards_operations_endpoint(
+        &self,
+        service_index: usize,
+        user_farms_pairs_to_claim: MultiValueEncoded<MultiValue2<AddressId, ManagedVec<AddressId>>>,
+    ) {
+        require!(service_index <= 1, "Invalid service index");
+        let energy_threshold = self.energy_threshold().get();
+        let premium_service = service_index == 1;
+
+        if premium_service {
+            require!(energy_threshold > 0, "Invalid energy threshold");
```

Comment:
premium check is not needed here. Bot executed these for the users and that's it.
Energy threshold has to be checked only when subtracting payments from users.